temporal-sdk/src/test/java/io/temporal/workflow/updateTest/UpdateWithStartTest.java

```diff
@@ -0,0 +1,542 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.workflow.updateTest;
+
+import static org.junit.Assert.*;
+
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import io.temporal.api.common.v1.WorkflowExecution;
+import io.temporal.api.enums.v1.WorkflowIdConflictPolicy;
+import io.temporal.api.enums.v1.WorkflowIdReusePolicy;
+import io.temporal.client.*;
+import io.temporal.testing.internal.SDKTestOptions;
+import io.temporal.testing.internal.SDKTestWorkflowRule;
+import io.temporal.worker.WorkerOptions;
+import io.temporal.workflow.*;
+import io.temporal.workflow.shared.TestMultiArgWorkflowFunctions;
+import io.temporal.workflow.shared.TestWorkflows;
+import java.util.*;
+import java.util.concurrent.ExecutionException;
+import java.util.function.BiFunction;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class UpdateWithStartTest {
+
```

Comment:
Do we have any test here that tests the update was executed before the main workflow method? Maybe I just missed it.

Reply:
I'm not sure I follow; why would that be an Update-with-Start-specific test? Isn't that already covered by the Update-specific tests?

Reply:
Because in the past there was not a good way to guarantee the update was in the first workflow task

Reply:
My main concern is how do you know the test server is sending the update in the first workflow task when using update-with-start?

Reply:
👍 Okay, I see what you mean.

Reply:
Okay, so I think the `startAndSendUpdateTogetherWithNullUpdateResult` test covers this. The workflow is

```
public static class TestUpdatedWorkflowImpl implements TestWorkflows.TestUpdatedWorkflow {

    private String state;

    @Override
    public String execute() {
      return state;
    }

    @Override
    public void update(String arg) {
      this.state = arg;
    }
  }
```

And it asserts that the workflow result equals the update input. If the update _wasn't_ executed first, it would return `null` instead and the test fails.